### PR TITLE
Change synonyms internal user permissions to fix synonyms API analyzers reloading

### PR DESCRIPTION
**Pull request by carlosdelest**
Synonyms internal user does not have enough permissions to reload analyzers on indices, as this needs the `manage` privilege.

This was previously not spotted because tests that involved reloading were located in the `analysis-common` module, which is not tested with security. Moved these tests as part of the synonyms REST tests so security is checked.

Adding `manage` privilege to the internal user so analyzer reloading is done correctly.

This work will need to be done before https://github.com/elastic/elasticsearch/pull/97622 so reloading returns the right permissions.

### CODE REVIEW

File: modules/analysis-common/build.gradle
Comment (carlosdelest) [inline modules/analysis-common/build.gradle:22]: Removed tests from analysis-common, so Synonyms API is no longer needed

File: rest-api-spec/build.gradle
Comment (carlosdelest) [inline rest-api-spec/build.gradle:40]: Add `analysis-common` to REST tests so synonyms analyzers are available for tests

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/synonyms/80_synonyms_from_index.yml
Comment (carlosdelest) [inline rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/synonyms/80_synonyms_from_index.yml:4]: This test needs to be bumped in version as well because of bwc tests

File: x-pack/qa/core-rest-tests-with-security/build.gradle
Comment (carlosdelest) [inline x-pack/qa/core-rest-tests-with-security/build.gradle:9]: ... also needed in tests with security

**Review by kderusso — COMMENTED** [review]
The changes LGTM, but for my education - are there any issues with issuing this manage privilege? Could someone use this user to manage indices they shouldn't have access to? (It doesn't _seem_ to be the case because it's an internal user, but I'd like to know a bit more?)

**Comment by carlosdelest** [comment]
> are there any issues with issuing this manage privilege? Could someone use this user to manage indices they shouldn't have access to?

The internal user is used for the synonyms related operations, using the OriginSettingClient. So it's possible that a user with manage_cluster privileges do a reload_search_analyzers operation that they ordinarily could not do in case they don't have the index privileges.

As this is something we control, and that we want to happen, I would say it's safe. The internal user can't be used for external operations. Index operations performed are not destructive as well.

We can potentially refine this in the future. But given the current nature of the operations performed, and that we want to reload all analyzers that use the synonym set, I believe it's a good usage.

**Review by kderusso — APPROVED** [review]

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/user/InternalUsers.java
Comment (mayya-sharipova) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/user/InternalUsers.java:183]: Is it possible to give less permission that `manage`? Can `read`  and `view_index_metadata` permissions work?

**Review by mayya-sharipova — CHANGES_REQUESTED** [review]
Thanks @carlosdelest for fixing this. Overall LGTM, just wondering if we can get away with less privilege.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/user/InternalUsers.java
Comment (carlosdelest) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/user/InternalUsers.java:183]: I checked doing the following setup:

`PUT /_security/role/test-role`
```json
{
  "cluster": [],
  "indices": [
    {
      "names": ["test-index"],
      "privileges": ["read"]
    }
  ],
  "applications": [],
  "run_as": []
}
```

`POST /_security/user/test-user`
```json
{
  "password" : "testtest",
  "roles" : ["test-role"]
}
```

`curl -u test-user:testtest -X POST http://localhost:9200/test-index/_reload_search_analyzers`
```json
{
  "error": {
    "root_cause": [
      {
        "type": "security_exception",
        "reason": "action [indices:admin/reload_analyzers] is unauthorized for user [test-user] with effective roles [test-role] on indices [test-index], this action is granted by the index privileges [manage,all]"
      }
    ],
    "type": "security_exception",
    "reason": "action [indices:admin/reload_analyzers] is unauthorized for user [test-user] with effective roles [test-role] on indices [test-index], this action is granted by the index privileges [manage,all]"
  },
  "status": 403
}
```
I checked using `view_index_metadata`, same result.

Do you see this as a concern that we should raise?

Comment (mayya-sharipova) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/user/InternalUsers.java:183]: Thanks for checking, Carlos. Let's have `manage` role then.

**Review by mayya-sharipova — APPROVED** [review]
@carlosdelest Thanks for fixing this! LGTM!

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/user/InternalUsers.java
Comment (javanna) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/user/InternalUsers.java:183]: have we checked with the security team about permissions?

**Comment by javanna** [comment]
heya @carlosdelest could we have a more descriptive title for this PR. I understand it relates to permissions, in that case should it mention permissions in its title? Thanks!

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/user/InternalUsers.java
Comment (carlosdelest) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/user/InternalUsers.java:183]: I'm checking with the security team in [this slack conversation](https://elastic.slack.com/archives/CA6N3HDF1/p1689328457266199). I had assumed we were ok for the [reasons stated above](https://github.com/elastic/elasticsearch/pull/97621#issuecomment-1633002289), which might have been wrong. 

Thanks for the ping, Luca!

**Pull request merged**
